File: TwoWayDemo/app/src/main/java/com/anushka/androidtutz/twowaydemo/MainActivity.java
package com.anushka.androidtutz.twowaydemo;

import android.databinding.DataBindingUtil;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

import com.anushka.androidtutz.twowaydemo.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding activityMainBinding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        activityMainBinding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        activityMainBinding.setContact(new Contact("Alex", "alex@gmail.com"));
    }
}

File: DIDemo/app/src/main/java/com/anushka/didemo/MainActivity.kt
package com.anushka.didemo

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.anushka.didemo.databinding.ActivityMainBinding
import javax.inject.Inject

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    @Inject
    lateinit var smartPhone: SmartPhone

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)

        /*DaggerSmartPhoneComponent.create()
            .inject(this)*/

        (application as SmartPhoneApplication).smartPhoneComponent.inject(this)

        smartPhone.makeACallWithRecording()
    }
}
